src/scope.ml

```diff
@@ -243,8 +241,9 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
       | (Some x::l,d,i)::xs ->
           let v = Bindlib.new_var mkfree x.elt in
           let a = scope_domain env d in
-          let env = if x.elt = "_" then env else (x.elt,(v,a))::env in
-          let t = aux env ((l,d,i)::xs) in
+          let t = aux ((x.elt,(v,a)) :: env) ((l,d,i) :: xs) in
```

Comment:
As I suggested in a previous comment (https://github.com/Deducteam/lambdapi/issues/292#issuecomment-558723937), I am not sure that this is a good idea: this impacts unification a lot. Variables that start with _ should be filtered out before generating new metavariables. Is it something that you do?

Reply:
Here I just removed dead code, the wildcard case is handled in the `None` branch.